sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -695,8 +695,14 @@ class Dataset[T] private[sql](
   // defined on a derived column cannot referenced elsewhere in the plan.
   def withWatermark(eventTime: String, delayThreshold: String): Dataset[T] = withTypedPlan {
     val parsedDelay =
-      Option(CalendarInterval.fromString("interval " + delayThreshold))
-        .getOrElse(throw new AnalysisException(s"Unable to parse time delay '$delayThreshold'"))
+      try {
+        CalendarInterval.fromCaseInsensitiveString(delayThreshold)
+      } catch {
+        case e: IllegalArgumentException =>
+          throw new AnalysisException(
+            s"Unable to parse time delay '$delayThreshold'",
+            cause = Some(e))
```

Comment:
This is good because this prevents the exception type change. I'm wondering if we need to propagate `cause` here  because previously we didn't. And, for the other instance in this PR, the exception messages are not the same.

Reply:
> I'm wondering if we need to propagate `cause` here because previously we didn't.

Previously it was just a `null` return value and there was no `cause`. Since we now get a `cause`, it's better to propagate it so that the error will be clearer.

> And, for the other instance in this PR, the exception messages are not the same.

This is something we need to pay to unify all usages. At least, the exception type doesn't change.

Reply:
Got it. Thanks!